src/rust/engine/src/fs.rs

```diff
@@ -636,26 +635,26 @@ impl fmt::Debug for Snapshot {
 }
 
 /**
- * A facade for the snapshot directory, which is currently thrown away at the end of each run.
+ * A facade for the snapshot directory, which lives under the pants workdir.
  */
 pub struct Snapshots {
-  temp_dir: TempDir,
+  snapshots_dir: PathBuf,
   next_temp_id: atomic::AtomicUsize,
 }
 
 impl Snapshots {
-  pub fn new() -> Result<Snapshots, io::Error> {
+  pub fn new(snapshots_dir: PathBuf) -> Result<Snapshots, io::Error> {
+    fs::create_dir_all(snapshots_dir.to_owned())?;
```

Comment:
nit: `to_owned` implies copying the path, but `create_dir_all` takes anything that matches `AsRef<Path>`... ie, anything that can be viewed as `&Path` (which would be a reference, rather than a copy).

So, if removing the `.to_owned()` doesn't "just work" here (I think it might?) you'd want to use `.as_path()` instead.